### PR TITLE
tests for web notification consumer

### PR DESCRIPTION
**Pull request by aquibbaig**
Tried writing some tests for the web notification consumer. @imphil @agathver Please suggest if I have missed something on this.

Since I found that protected/private methods should not be tested, I did some refactoring to the existing code itself to test out the `handle()` function

### CODE REVIEW

File: site/tests/RepoCrawler/WebNotificationConsumerTest.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        $this->assertEquals('new_project', $this->notification->getType());
+        $this->assertEquals('test', $this->notification->getRecipient()->getUsername());
+    }
+
+    /**
+     * @test
+     *
+     * Tests if a notification is added to database
+     */
+    public function test_if_notification_is_added()
```
Comment (imphil) [inline site/tests/RepoCrawler/WebNotificationConsumerTest.php]: use lowerCamelCase() for function names.

File: site/tests/RepoCrawler/WebNotificationConsumerTest.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        $mockLogger = $this->createMock(LoggerInterface::class);
+        $mockNotificationManager = $this->getMockBuilder(NotificationManager::class)
+            ->setMethods(['addNotification'])
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $webNotificationConsumer = new WebNotificationConsumer($mockLogger, $mockNotificationManager);
+        $webNotificationConsumer->setNotification($this->notification);
```
Comment (imphil) [inline site/tests/RepoCrawler/WebNotificationConsumerTest.php]: That doesn't test what is really happening. What you should do instead is run the execute() function with a serialized notification object.

File: site/tests/RepoCrawler/WebNotificationConsumerTest.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        // Test if handle() doesn't requeue notifications in RabbitMQ
+        $this->assertEquals(true, $webNotificationConsumer->handle());
```
Comment (imphil) [inline site/tests/RepoCrawler/WebNotificationConsumerTest.php]: Don't put the function call into the assert

File: site/tests/RepoCrawler/WebNotificationConsumerTest.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+        // Test if notification is added to database
+        $mockNotificationManager->expects($this->once())
+            ->method('addNotification');
+
+        $webNotificationConsumer->handle();
```
Comment (imphil) [inline site/tests/RepoCrawler/WebNotificationConsumerTest.php]: What's that for? You already called handle() before.

File: site/tests/RepoCrawler/WebNotificationConsumerTest.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    }
+}
+
+?>
```
Comment (imphil) [inline site/tests/RepoCrawler/WebNotificationConsumerTest.php]: Leave that out.

File: site/tests/RepoCrawler/WebNotificationConsumerTest.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        // Test if handle() doesn't requeue notifications in RabbitMQ
+        $this->assertEquals(true, $webNotificationConsumer->handle());
+
+        // Test if notification is added to database
+        $mockNotificationManager->expects($this->once())
+            ->method('addNotification');
+
+        $webNotificationConsumer->handle();
+    }
+}
```
Comment (imphil) [inline site/tests/RepoCrawler/WebNotificationConsumerTest.php]: The test_if_notification_is_added() test is incomplete, it doesn't explore all possible paths the function can take.
- What happens if shouldHandle() returns true or false? How can you test both cases?
- In which cases returns handle() a 0 or non-zero return value? What can do you to test both cases?

Comment (aquibbaig) [reply site/tests/RepoCrawler/WebNotificationConsumerTest.php]: @imphil For testing both shouldHandle and handle, we need to set them as public first?

Comment (aquibbaig) [reply site/tests/RepoCrawler/WebNotificationConsumerTest.php]: Else we need to have a public method that calls them, what's the better option?

Comment (aquibbaig) [reply site/tests/RepoCrawler/WebNotificationConsumerTest.php]: @imphil I just made the possible tests, please have a look at it

File: site/tests/RepoCrawler/WebNotificationConsumerTest.php [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+        $this->assertEquals('new_project', $this->notification->getType());
+        $this->assertEquals('test', $this->notification->getRecipient()->getUsername());
+    }
+
+    /**
+     * Called after every test
+     */
+    public function tearDown()
+    {
+        $this->notification = [];
```
Comment (imphil) [inline site/tests/RepoCrawler/WebNotificationConsumerTest.php]: Why do you set the notification variable to an array?

File: site/tests/RepoCrawler/WebNotificationConsumerTest.php [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+    /**
+     * @test
+     *
+     * Test if notification of a specific
+     * type are handled
+     */
+    public function testIfMentionedTypesOfNotificationsAreHandled()
+    {
+
+
```
Comment (imphil) [inline site/tests/RepoCrawler/WebNotificationConsumerTest.php]: Remove empty lines here.

File: site/tests/RepoCrawler/WebNotificationConsumerTest.php [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+    public function tearDown()
+    {
+        $this->notification = [];
+    }
+
+    /**
+     * @test
+     *
+     * Test if notification of a specific
+     * type are handled
```
Comment (imphil) [inline site/tests/RepoCrawler/WebNotificationConsumerTest.php]: make that one line

**Comment by aquibbaig** [comment]
@imphil Fixed!

File: site/tests/RepoCrawler/WebNotificationConsumerTest.php (hunk 7 above)
Comment (aquibbaig) [reply site/tests/RepoCrawler/WebNotificationConsumerTest.php]: I have set it to null now

**Pull request merged**